examples/step-57/doc/intro.dox

```diff
@@ -0,0 +1,319 @@
+\left(\begin{array}{cc} \tilde{A}^{-1} & 0 \\ 0 & I \end{array}\right)
+\left(\begin{array}{cc} I & -B^T \\ 0 & I \end{array}\right)
+\left(\begin{array}{cc} I & 0 \\ 0 & \tilde{S}^{-1} \end{array}\right).
+@f}
+
+Here two inexact solvers will be needed for $\tilde{A}^{-1}$ and
+$\tilde{S}^{-1}$, respectively(see [1]). Since the pressure mass
+matrix is symmetric and positive definite,
+CG with ILU as a preconditioner is appropriate to use for $\tilde{S}^{-1}$. For simplicity, we use 
+the direct solver UMFPACK for $\tilde{A}^{-1}$. The last ingredient is a sparse
+matrix-vector product with $B^T$. Instead of computing the matrix product
+in the augmented Lagrangian term in $\tilde{A}$, we assemble Grad-Div stabilization
+$(\nabla \cdot \phi _{j}, \nabla \cdot \phi _{i}) \approx (B^T
+M_p^{-1}B)_{ij}$, as explained in [2].
+
+<h3> Test Case </h3>
+
+Here we use the Lid driven cavity flow as our test case, see [3] for details. 
+The computational domain is the unit square and the right-hand side
+$f=0$. The boundary condition is
+
+@f{eqnarray*} 
+(u(x, y), v(x,y)) &=& (1,0)   
+  \qquad\qquad \textrm{if}\ y = 1 \\
+  (u(x, y), v(x,y)) &=& (0,0)
+  \qquad\qquad \textrm{else}.
+@f}
+
+When solving this problem, the error consists of the nonlinear error (from
+Newton's iteration) and the discrete error (depending on mesh size). The
+nonlinear part decreases with each Newton iteration and the discrete error
+reduces with mesh refinement. In this example, the solution from the coarse
+mesh is transferred to successively finer meshes and used as an initial
+guess. Therefore, the nonlinear error is always brought below the tolerance of
+Newton's iteration and the discrete error is reduced with each mesh
+refinement.
+
+Inside the loop, we involve three solvers: one for $\tilde{A}^{-1}$,
+one for $M_p^{-1}$ and one for $Gx=b$. The first two
+solvers are invoked in the preconditioner and the outer solver gives us
+the update term. Overall convergence is controlled by the nonlinear residual 
+and Newton's method does not require an exact Jacobian, so for the outer 
+linear solver we employ FGMRES with a relative tolerance of only 1e-4. 
+As described in step-22, the inner linear solves are also not required 
+to be done very accurately. Here we use CG with a relative 
+tolerance of 1e-4 for the mass matrix. As expected, we still see convergence 
+of the nonlinear residual down to 1e-14. Also, we use a simple line
+search algorithm for globalization of the Newton method.
+
+The cavity reference values for Re=400 and Re=7500 are from [4] and [5],
+respectively. Even though we can still find a solution for Re=10000 and the
+references contain results for comparison, we limit our discussion here to
+Re=7500. This is because the solution is no longer stationary starting around
+Re=8000 but it becomes periodic, see [7] for details.
+
+<h3> Reference </h3>
+<ol>
+
+  <li>  An Augmented Lagrangian-Based Approach to the Oseen Problem, M. Benzi and M. Olshanskii
```

Comment:
I think it would be useful to provide complete references here, with year and journal